store/src/java/com/zimbra/cs/service/FileUploadServlet.java

```diff
@@ -222,6 +223,10 @@ public static MimeType getMimeType(FileItem fileItem) {
                 } else {
                     mediaType = detector.detect(stream, metadata);
                 }
+                // special-case text/xml to avoid detection of text/plain
+                if (MimeConstants.CT_TEXT_XML.equals(contentType) || MimeConstants.CT_TEXT_XML_LEGACY.equals(contentType)) {
```

Comment:
if mediaType is already xml and contentType is also xml, then unnecssarily we will again be parsing this content. So shall we handle mediaType clause too here in if statement (if mediaType != xml)
Cc: @dasiyogesh

Reply:
but we will get media type is xml only with tika custom mime file , and it is not able to detect